spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/DealHeaderUtil.java

```diff
@@ -59,35 +60,29 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(DealHeaderUtil.class);
 
   public static void addAKSKHeader(HttpUriRequest httpRequest,
-      SSLConfig sslConfig) {
+      AkSkConfig akSkConfig) {
     AuthHeaderUtils authHeaderUtils = AuthHeaderUtils.getInstance();
     Map<String, String> headerMap = authHeaderUtils.genAuthHeaders();
-    for (Map.Entry<String, String> entry : headerMap.entrySet()) {
-      httpRequest.addHeader(entry.getKey(), entry.getValue());
+    if (akSkConfig.isAkSkEmpty() && !CollectionUtils.isEmpty(headerMap)) {
+      httpRequest.addHeader(X_SERVICE_AK, headerMap.get(X_SERVICE_AK));
+      httpRequest.addHeader(X_SERVICE_SHA_AKSK, headerMap.get(X_SERVICE_SHA_AKSK));
+    } else {
+      httpRequest.addHeader(X_SERVICE_AK, akSkConfig.getAccessKey());
+      httpRequest.addHeader(X_SERVICE_SHA_AKSK, encode(akSkConfig));
     }
-    if (isHeaderMapEmpty(headerMap) && isSSLConfigNotEmpty(sslConfig)) {
-      httpRequest.addHeader(X_SERVICE_AK, sslConfig.getAccessKey());
-      httpRequest.addHeader(X_SERVICE_SHA_AKSK,
-          encode(sslConfig));
-      httpRequest.addHeader(X_SERVICE_PROJECT, sslConfig.getProject());
+    if (akSkConfig.isProjectEmpty() && !CollectionUtils.isEmpty(headerMap)) {
+      httpRequest.addHeader(X_SERVICE_PROJECT, headerMap.get(X_SERVICE_PROJECT));
+    } else {
+      httpRequest.addHeader(X_SERVICE_PROJECT, akSkConfig.getProject());
     }
   }
 
-  public static boolean isHeaderMapEmpty(Map<String, String> headerMap) {
-    return headerMap == null || headerMap.size() == 0;
-  }
-
-  public static boolean isSSLConfigNotEmpty(SSLConfig sslConfig) {
-    return sslConfig.getAccessKey() != null && sslConfig.getSecretKey() != null && sslConfig.getProject() != null;
-  }
-
-  private static String encode(SSLConfig sslConfig) {
-
-    if ("ShaAKSKCipher".equalsIgnoreCase(sslConfig.getAkskCustomCipher())) {
-      return sslConfig.getSecretKey();
+  private static String encode(AkSkConfig akSkConfig) {
+    if ("ShaAKSKCipher".equalsIgnoreCase(akSkConfig.getAkskCustomCipher())) {
```

Comment:
ShaAKSKCipher can be a constant in AkSkConfig

Reply:
done